app/queries/admin/users_query.rb

```diff
@@ -31,6 +46,22 @@ def self.search_relation(relation, search)
       relation.where(QUERY_CLAUSE, search: "%#{search.strip}%")
     end
 
+    def self.filter_joining_date(relation:, joining_start:, joining_end:, date_format:)
+      ui_formats_to_parse_format = {
+        "DD/MM/YYYY" => "%d/%m/%Y",
+        "MM/DD/YYYY" => "%m/%d/%Y"
+      }
+      parse_format = ui_formats_to_parse_format[date_format]
```

Comment:
We're going to have a downstream exception if we don't have a format match (see below).  I recommend that you change the above to what follows; this will raise a `KeyError` because someone munged the request parameters:

```suggestion
      parse_format = ui_formats_to_parse_format.fetch(date_format)
```


```ruby
[3] pry(main)> DateTime.strptime("09/03/2022", nil)
TypeError: no implicit conversion of nil into String
from (pry):3:in `strptime'
```

Reply:
This is much better - thanks!